### DISCUSSION TITLE
renovate-config-validator: add write migrations mode

### DISCUSSION THREAD

**Discussion by massongit** [renovatebot/renovate#30579] [Suggest an Idea]
### Tell us more.

I think it would be nice if there was an option in renovate-config-validator to write migrations to a real file if necessary.

**Comment by HonkingGoose** [comment]
We have an _experimental_ config option: [`configMigration`](https://docs.renovatebot.com/configuration-options/#configmigration) that you can use to get config migration PRs from Renovate. Set `"configMigration": true,` in your Renovate config file. Note that you may have to run Prettier or fix-up some whitespace/newline problems yourself though!

We plan to have Renovate create "config migration PRs" by default. See this issue for a list of remaining todos:

- #16359

**Edit:** I'll let a maintainer respond if they want to make it so the config validator can write the migrations into the file.

**Reply by massongit** [reply]
Thank you for your answer!
I will watch that Issue.

**Reply by HonkingGoose** [reply]
More users want the config validator to write to the file:

- #30622

I'm re-opening this discussion so a maintainer can respond. 🙂

**Reply by Finkregh** [reply]
Thanks, somehow my search did not lead me here.
If this is implemented we could also extend the pre-commits as in https://github.com/renovatebot/pre-commit-hooks/pull/2279, that it directly updates the configs.

**Comment by rarkins** [comment]
Converted to feature request: https://github.com/renovatebot/renovate/issues/30989
